tests-arquillian/src/test/java/org/jboss/weld/tests/interceptors/producer/InterceptionFactoryTest.java

```diff
@@ -138,27 +138,55 @@ public void testFactoryFromGenericInterface(@Produced SomeGenericInterface<List<
         assertEquals(2, Producer.INVOCATIONS.size());
     }
 
-    @Inject
-    @Any
-    Instance<Object> beanInstance;
+    @Test
+    public void testFactoryBackedByUnproxyableImpl(@Produced ProxyableInterface bean) {
+        // this test adds class level Monitor and method level Hello bindings
+        Producer.reset();
+        assertEquals("Hello " + UnproxyableImpl.class.getSimpleName(), bean.ping());
+        assertEquals(1, Producer.INVOCATIONS.size());
+    }
+
+    @Test
+    public void testFactoryFromInterfaceWithMethodAnnotation(@Produced ProxyableInterfaceWithMethodAnnotation bean) {
+        // this test adds class level Monitor and method level Hello bindings
+        Producer.reset();
+        assertEquals("Hello " + UnproxyableInterfaceWithMethodAnnotationImpl.class.getSimpleName(), bean.ping());
+        assertEquals(1, Producer.INVOCATIONS.size());
+    }
 
     @Test
-    public void testFactoryFromInterfaceWithMethodAnnotation() {
-        try {
-            beanInstance.select(UninterestingInterface1.class, Produced.Literal.INSTANCE).get().ping();
-            Assert.fail();
-        } catch (IllegalStateException e) {
-            // expected
-        }
+    public void testFactoryFromInterfaceWithClassAnnotation(@Produced ProxyableInterfaceWithClassAnnotation bean) {
+        // this test adds class level Monitor and method level Hello bindings
+        Producer.reset();
+        assertEquals("Hello " + UnproxyableInterfaceWithClassAnnotationImpl.class.getSimpleName(), bean.ping());
+        assertEquals(1, Producer.INVOCATIONS.size());
     }
 
     @Test
-    public void testFactoryFromInterfaceWithTypeAnnotation() {
-        try {
-            beanInstance.select(UninterestingInterface2.class, Produced.Literal.INSTANCE).get().ping();
-            Assert.fail();
-        } catch (IllegalStateException e) {
-            // expected
-        }
+    public void testFactoryFromGenericInterfaceStructure(@Produced InterfaceWithGenericsB<String, Integer> bean) {
+        Producer.reset();
+        bean.ping(); // this doesn't trigger anything as we don't inherit annotations ATM
```

Comment:
Class-level annotation is not applied to a predecessor in interface hierarchy. Spec doesn't define this, it's up to us.
Ideas on how it should behave?

Reply:
I'd say we should follow the JLS and `@Inherited`, i.e. annotations are not inherited by default.